### ISSUE TITLE
About the warmup process

### ISSUE THREAD

**Issue by RIU-13** [hhc1997/L2RM#3]
Hello~
Why loss is decreasing while the accuracy is decreasing at the same time ? After 1 epoch, the accuracy is about 70%, then it drops as the epoch increases. Maybe this doesn't matter?

**Comment by hhc1997** [comment]
Hi @RIU-13 
What's the noise ratio in your experiment?
Does the "1 epoch" you mentioned mean during or after warmup?

**Comment by RIU-13** [comment]
> Hi @RIU-13 What's the noise ratio in your experiment? Does the "1 epoch" you mentioned mean during or after warmup?

For noise ratio=0, after 1 epoch during warmup, the accuracy is 0.72, then it drops like 0.71, 0.70.
For noise ratio=0.2, after 1 epoch during warmup, the accuracy is 0.70, then it drops like 0.69, 0.68.
I am new to cross-modal matching task, and I wonder the noise file is generated during training process or downloaded from the website you give? In your code, the noise index file can be generated during warmup.

**Comment by hhc1997** [comment]
Hi, @RIU-13 
Could you please provide the training log?
If you use a specific noise file, the code will load the corresponding noisy setting, otherwise the code will generate a new noise file and use it.

**Comment by RIU-13** [comment]
> Hi, @RIU-13 Could you please provide the training log? If you use a specific noise file, the code will load the corresponding noisy setting, otherwise the code will generate a new noise file and use it.

OK. I am trying to apply your method to cross-modal retrieval task in the total MirFlickr 25K dataset and the evaluation metric is MAP.
In my task, the noise file should be generated rather than specified.
####training log####

[1/3] Warmup model                                                                                                                                                                               
Warmup Step[  0/141]    batch  5.521 ( 5.521)   data  0.000 ( 0.000)    loss 2.0997e+01 (2.0997e+01)                                                                                             
Warmup Step[100/141]    batch  0.190 ( 0.242)   data  0.000 ( 0.000)    loss 1.7377e+01 (1.8829e+01)                                                                                             
                                                                                                                                                                                                 
 Validattion ...                                                                                                                                                                                 
Test                                                                                                                                                                                             
Img2Txt: 0.665626        Txt2Img: 0.658687       Avg: 0.662157                                                                                                                                   
[2/3] Warmup model                                                                                                                                                                               
Warmup Step[  0/141]    batch  1.211 ( 1.211)   data  0.000 ( 0.000)    loss 1.5499e+01 (1.5499e+01)                                                                                             
Warmup Step[100/141]    batch  0.192 ( 0.200)   data  0.000 ( 0.000)    loss 1.5213e+01 (1.5142e+01)                                                                                             
                                                                                                                                                                                                 
 Validattion ...                                                                                                                                                                                 
Test              Img2Txt: 0.654305        Txt2Img: 0.646235       Avg: 0.65027
[3/3] Warmup model
Warmup Step[  0/141]    batch  1.511 ( 1.511)   data  0.000 ( 0.000)    loss 1.2133e+01 (1.2133e+01)
Warmup Step[100/141]    batch  0.191 ( 0.203)   data  0.000 ( 0.000)    loss 1.2856e+01 (1.2373e+01)

 Validattion ...
Test
Img2Txt: 0.643086        Txt2Img: 0.633415       Avg: 0.63825
=> load noisy index from mir_0.2_1.npy
train_all data has a size of 18015
Epoch 1 / 100
Split dataset ...
Correct data acc: 0.9133767727098505
Total data acc: 0.91834582292534
=> load noisy index from mir_0.2_1.npy
labeled data has a size of 15654
=> load noisy index from mir_0.2_1.npy
unlabeled data has a size of 2361
Model training ...

 Training...
Training Step[  0/122]  batch  2.387 ( 2.387)   loss 4.2583e+01 (4.2583e+01)    cost_losses 2.0482e+00 (2.0482e+00)
Training Step[100/122]  batch  0.361 ( 0.455)   loss 5.1181e+01 (5.1494e+01)    cost_losses 6.6151e+00 (5.5583e+00)

 Validattion ...
Test
Img2Txt: 0.588021        Txt2Img: 0.611555       Avg: 0.599788
Epoch 2 / 100
Split dataset ...
Correct data acc: 0.5760336370007008
Total data acc: 0.21199000832639467
=> load noisy index from mir_0.2_1.npy
labeled data has a size of 1427
=> load noisy index from mir_0.2_1.npy
unlabeled data has a size of 16588

Model training ...

 Training...
Training Step[ 0/11]    batch  2.585 ( 2.585)   loss 5.2278e+01 (5.2278e+01)    cost_losses 1.8669e+00 (1.8669e+00) 
Validattion ...
Test
Img2Txt: 0.581776        Txt2Img: 0.607203       Avg: 0.594489

**Comment by hhc1997** [comment]
Hi, @RIU-13
From the experimental log, the model seems to be overfitting qucikly. You should tune some parameters to suit your task, e.g., reducing the learning rate.
By the way, Do you change the backbone model (i.e., SGR and SAF)?

**Comment by RIU-13** [comment]
> Hi, @RIU-13 From the experimental log, the model seems to be overfitting qucikly. You should tune some parameters to suit your task, e.g., reducing the learning rate. By the way, Do you change the backbone model (i.e., SGR and SAF)?

Yes, I change the backbone model. The image net connects the pretrained vgg model and several linear layers, and the text net simply consists of several linear layers, which is widely used in my task. 
So do I need to change the 'method' in SGRAF (i.e., RCE and Sym_KLDIV) or just tune the parameters ? In my analysis, these methods are supposed to work in my task since they are based on triplet loss and InfoNCE.

**Comment by hhc1997** [comment]
Hi, @RIU-13
Just tune the parameters. Your backbone models are pretty simple. The provided parameters are based on SGR and SAF.

**Comment by RIU-13** [comment]
OK. Many thanks for your patient replying 🤍
